pkg/nimble/netif/nimble_netif.c

```diff
@@ -442,16 +429,22 @@ static int _on_gap_master_evt(struct ble_gap_event *event, void *arg)
                 _notify(handle, NIMBLE_NETIF_ABORT_MASTER, addr);
                 return 0;
             }
+
             _on_gap_connected(conn, event->connect.conn_handle);
             conn->state |= NIMBLE_NETIF_GAP_MASTER;
 
-            struct os_mbuf *sdu_rx = os_mbuf_get_pkthdr(&_mbuf_pool, 0);
-            /* we should never run out of buffer space... */
-            assert(sdu_rx != NULL);
+            struct os_mbuf *sdu_rx = os_msys_get_pkthdr(
+                                            MYNEWT_VAL_BLE_L2CAP_COC_MPS, 0);
+            /* in the rare case that we run out of buffers, we close the new
+             * connection right away */
+            if (sdu_rx == NULL) {
+                ble_gap_terminate(handle, BLE_ERR_REM_USER_CONN_TERM);
+                break;
+            }
```

Comment:
Why did this change from "we should never run out of buffer space..." to "in the rare case that we run out of buffers", just future proofing?

Reply:
because before we used buffer space that was solely allocated and sized for this specific purpose while now we use shared buffers that could potentially be exhausted by other modules